### DISCUSSION TITLE
asyncComputed not working after route change

### DISCUSSION THREAD

**Discussion by creart-jetsai** [vueuse/vueuse#2034] [Q&A]
I use asyncComputed inside pinia store setup as below.

```js
// chatStore.ts
import { ref } from 'vue';
import { defineStore } from 'pinia';
import { asyncComputed } from '@vueuse/core';

export const chatStore = defineStore('chat', () => {
    const chatId = ref('')
   
    const chat = asyncComputed(() => {
        if(!chatId.value) return null
        
        const res =  await api.getChatMessages(chatId.value)
        return res.data.messages
    }, null)

    return {
        chatId,
        chat,
    }
})
```

It works well in my component. But it's no longer triggered after I go to another route and get back.
I checked that `chatId` did change but async callback never executed again.

**Comment by curtgrimes** [comment]
Hi @creart-jetsai. I'm having trouble reproducing the issue you're experiencing. I took your code and was able to see `asyncComputed()` rerunning when `chatId.value` changes. In this example I made in StackBlitz you can see the third parameter in `chat` is the `chatId` itself, and it changes when you type in a different `chatId` into the text box: https://stackblitz.com/edit/vue3-script-setup-with-vite-ndx5se?file=src%2FchatStore.ts

I did notice that in your example you are missing an `async` but I don't know if that would cause the issue you were having:

```diff
// chatStore.ts
import { ref } from 'vue';
import { defineStore } from 'pinia';
import { asyncComputed } from '@vueuse/core';

export const chatStore = defineStore('chat', () => {
    const chatId = ref('')

-   const chat = asyncComputed(() => {   
+   const chat = asyncComputed(async () => {
        if(!chatId.value) return null
        
        const res =  await api.getChatMessages(chatId.value)
        return res.data.messages
    }, null)

    return {
        chatId,
        chat,
    }
})
```

If that doesn't fix your issue, can you create a minimal reproduction on something like StackBlitz or edit mine to reproduce it? Thanks!

**Comment by creart-jetsai** [comment]
@curtgrimes Thanks for your reply

https://stackblitz.com/edit/vueuse-asynccomputed-issue?file=README.md

I create this reproduction for you to inspect the error

Follow these steps:

1. open the link and see you are in `foo` page and the chat text is "foo"
2. click `Go To Bar` to Bar page and see the chat text change to "chat"
3. click `Go To Foo` to Foo page and see the chat text **not** change to "foo" as expected
